cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/artifact/ArtifactStore.java

```diff
@@ -531,6 +532,49 @@ public ArtifactData apply(DatasetContext<Table> context) throws Exception {
   }
 
   /**
+   * Update artifact properties using an update function. Functions will receive an immutable map.
+   *
+   * @param artifactId the id of the artifact to add
+   * @param updateFunction the function used to update existing properties
+   * @throws ArtifactNotFoundException if the artifact does not exist
+   * @throws IOException if there was an exception writing the properties to the metastore
+   */
+  public void updateArtifactProperties(final Id.Artifact artifactId,
+                                       final Function<Map<String, String>, Map<String, String>> updateFunction)
+    throws ArtifactNotFoundException, IOException {
+    try {
+      boolean exists = metaTable.execute(new TransactionExecutor.Function<DatasetContext<Table>, Boolean>() {
+
+        @Override
+        public Boolean apply(DatasetContext<Table> context) throws Exception {
+          Table table = context.get();
+
+          ArtifactCell artifactCell = new ArtifactCell(artifactId);
+          byte[] existingMetaBytes = table.get(artifactCell.rowkey, artifactCell.column);
+          if (existingMetaBytes == null) {
+            return false;
+          }
+
+          ArtifactData old = gson.fromJson(Bytes.toString(existingMetaBytes), ArtifactData.class);
+
+          ArtifactMeta updatedMeta = new ArtifactMeta(old.meta.getClasses(), old.meta.getUsableBy(),
+                                                      updateFunction.apply(old.meta.getProperties()));
+          ArtifactData updatedData = new ArtifactData(locationFactory.create(old.locationURI), updatedMeta);
+          // write artifact metadata
+          writeMeta(table, artifactId, updatedData);
```

Comment:
so this writes a lot more information than what we want updated right?. we just want to update the properties, which is serialized by this put ``table.put(artifactCell.rowkey, artifactCell.column, Bytes.toBytes(gson.toJson(data)));`` in ``writeMeta``method, whereas writeMeta does more than that including updating all `PluginClass` and `ApplicationClass` which we don't change.